From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Paul Gofman <pgofman@codeweavers.com>
Date: Thu, 22 Sep 2022 15:48:40 -0500
Subject: [PATCH] msvcrt: HACK: Disable SSE2 support for Indiana Jones and The
 Emperor Tomb.

CW-Bug-Id: #21330
Valve-commit: 2b1f2b43060aaa4003048d843f4fedcc9a7897ac
Picked: clean
Class: valve
Evidence: docs/evidence/2026-09-28-wine-proton-rebase.md
Offered-upstream: no
---
 dlls/msvcrt/math.c | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/dlls/msvcrt/math.c b/dlls/msvcrt/math.c
index 98d3a35..23c3df3 100644
--- a/dlls/msvcrt/math.c
+++ b/dlls/msvcrt/math.c
@@ -73,6 +73,15 @@ void msvcrt_init_math( void *module )
     sse2_supported = IsProcessorFeaturePresent( PF_XMMI64_INSTRUCTIONS_AVAILABLE );
 #if _MSVCR_VER <=71
     sse2_enabled = FALSE;
+    {
+        char sgi[64];
+
+        if (GetEnvironmentVariableA("SteamGameId", sgi, sizeof(sgi)) && !strcmp(sgi, "560430"))
+        {
+            sse2_supported = FALSE;
+            FIXME("HACK: disabling sse2 support in msvcrt.\n");
+        }
+    }
 #else
     sse2_enabled = sse2_supported;
 #endif
